### ISSUE TITLE
How can I track the progress of the download by JSON-RPC?

### ISSUE THREAD

**Issue by shangxinbo** [aria2/aria2#839]
I come to know that aria2 can print a readout to the console to show the progress of the downloads. look like this:
```
[#2089b0 400.0KiB/33.2MiB(1%) CN:1 DL:115.7KiB ETA:4m51s]
```
How can I track this progress info by JSON-PRC? Should I polling?

**Comment by tatsuhiro-t** [comment]
Yes, you can do polling to get progress.  Use aria2.tellStatus or aria2.tellActive.

**Comment by shangxinbo** [comment]
oh my good.if I polling to do that, the websocket is usefull to me.it's useless for websocket ,is it?

**Comment by tatsuhiro-t** [comment]
Those methods can be used with WebSocket.  Is that the answer you expected?

**Comment by shangxinbo** [comment]
I guess the websocket can push the info of progress for a download.I wonder what's the advantage for websocket to http?

**Comment by tatsuhiro-t** [comment]
WebSocket can push events such as download start, stop, etc but not progress (e.g., completed N%).
Please read the manual http://aria2.github.io/manual/en/html/aria2c.html#json-rpc-over-websocket

**Comment by shangxinbo** [comment]
How I wish I could get the progress to push, so I don't have to polling. In my opinion,push is more elegant than the poll, what do you think?Would you like to add this feature to the new version?

**Comment by shangxinbo** [comment]
I read the doc.I think download start is a trigger,so I needless a push.So do the stop trigger.And complete status can resolved by progress,Isn't it? So I think the push of events by now is useless.

**Comment by tatsuhiro-t** [comment]
If I understand your comment correctly, you'd like to say that progress push notification can cover the features that current event notification methods.  That's probably true for your use case.  Current push notifications are not useless, they have their own purposes, and besides that we don't have progress notification now.

**Comment by shangxinbo** [comment]
that is just one of my opinion. The most important in my case is real-time progress info. I mean the progress is important than the start/stop/complete event.
